app/src/main/java/rick_and_morty/data/repository/CharacterRepository.kt

```diff
@@ -0,0 +1,24 @@
+package rick_and_morty.data.repository
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import rick_and_morty.di.RickAndMortyApiRemoteDataSource
+import javax.inject.Inject
+import rick_and_morty.data.model.CharacterResultsDto
+
+
+class CharacterRepository @Inject constructor(
+    private val rickAndMortyApiRemoteDataSource: RickAndMortyApiRemoteDataSource
+){
+    
+   fun getCharacters(page: Int): Flow<List<CharacterResultsDto>> = flow {
+        emit(rickAndMortyApiRemoteDataSource.fetchRickAndMortyData(page).results!!)
+    }.flowOn(Dispatchers.Default)
```

Comment:
this is emitting exactly one value, meaning it can be a `suspend` function. Any reason it's a flow and not a suspend function?

also could we make `results` not null in the model to avoid the dangerous `!!`?

Reply:
done